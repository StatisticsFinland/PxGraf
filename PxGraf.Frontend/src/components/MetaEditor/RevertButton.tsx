import React from 'react';
import { useTranslation } from 'react-i18next';

import { IconButton, Tooltip} from '@mui/material';
import UndoIcon from '@mui/icons-material/Undo';

interface IRevertButtonProps {
    onClick: (value?: string) => void;
}

export const RevertButton: React.FC<IRevertButtonProps> = ({ onClick }) => {
    const { t } = useTranslation();

    return (
        <Tooltip title={t("editMetadata.discardChanges")}>
            <span>{/* span wrapper is needed to tooltip work when IconButton is disabled */}
                <IconButton
                    aria-label={t("editMetadata.discardChanges")}
                    onClick={() => onClick()}
                    edge="end"
                >
                    <UndoIcon />
                </IconButton>
            </span>
        </Tooltip>
    );
}

export default RevertButton;