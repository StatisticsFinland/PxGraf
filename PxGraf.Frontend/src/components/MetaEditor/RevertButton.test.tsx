import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import RevertButton from "./RevertButton";

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<RevertButton onClick={() => undefined} />);
        expect(asFragment()).toMatchSnapshot();
    });
});