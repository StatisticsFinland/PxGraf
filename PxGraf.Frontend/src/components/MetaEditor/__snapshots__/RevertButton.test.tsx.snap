// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <span
    aria-label="editMetadata.discardChanges"
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      aria-label="editMetadata.discardChanges"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="UndoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
        />
      </svg>
    </button>
  </span>
</DocumentFragment>
`;
